cosmo_tester/test_suites/bootstrap_based_tests/multi_network_test.py

```diff
@@ -101,11 +114,78 @@ def test_multiple_networks(managers,
     delete_manager(old_manager, logger)
 
     new_manager.use()
-    for hello in multi_network_hello_worlds[1:]:
+    for hello in multi_network_hello_worlds:
         hello.manager = new_manager
         hello.uninstall()
         hello.delete_deployment()
 
+    _add_new_network(new_manager, tmpdir, logger)
+    post_bootstrap_hello.verify_all()
+
+
+def _add_new_network(manager, tmpdir, logger):
+    logger.info('Adding network `{0}` to the new manager'.format(NETWORK_2))
+
+    local_metadata_path = str(tmpdir / 'certificate_metadata')
+    local_old_metadata_path = str(tmpdir / 'old_certificate_metadata')
+    remote_metadata_path = '/etc/cloudify/ssl/certificate_metadata'
+    private_ip = manager.private_ip_address
+
+    old_networks = deepcopy(manager.networks)
+    new_networks = deepcopy(manager.networks)
+
+    # `network_2` shouldn't be on the manager right now
+    old_networks.pop(NETWORK_2)
+
+    # This should add back `network_2` that we removed earlier, in the
+    # preconfigure callback
+    cert_metadata = {
+        'networks': new_networks,
+        'internal_rest_host': private_ip
+    }
+    with open(local_metadata_path, 'w') as f:
+        json.dump(cert_metadata, f)
+
+    with manager.ssh() as fabric_ssh:
+        logger.info('Validating old `certificate_metadata`...')
+        fabric_ssh.get(
+            remote_metadata_path, local_old_metadata_path, use_sudo=True
+        )
+        with open(local_old_metadata_path, 'r') as f:
+            old_metadata = yaml.load(f)
+
+        assert old_metadata['networks'] == old_networks
+
+        logger.info('Putting the new `certificate_metadata`...')
+        fabric_ssh.put(
+            local_metadata_path, remote_metadata_path, use_sudo=True
+        )
+
+        ip_setter_path = '/opt/cloudify/manager-ip-setter/'
+        restservice_python = '/opt/manager/env/bin/python'
+        mgmtworker_python = '/opt/mgmtworker/env/bin/python'
+        update_ctx_script = join(ip_setter_path, 'update-provider-context.py')
+        certs_script = join(ip_setter_path, 'create-internal-ssl-certs.py')
+
+        logger.info('Updating the provider context...')
+        fabric_ssh.sudo('{python} {script} --networks {networks} {ip}'.format(
+            python=restservice_python,
+            script=update_ctx_script,
+            networks=remote_metadata_path,
+            ip=private_ip
+        ))
+
+        logger.info('Recreating internal certs')
+        fabric_ssh.sudo('{python} {script} {ip}'.format(
```

Comment:
maybe let's also use explicit `--metadata filename` here instead of relying on the default, so that if a user reads this test as an example of how to do this, they understand that this file can be passed here